Applications/Spire/Source/Ui/ColorSelectorHexInputWidget.cpp

```diff
@@ -61,22 +61,28 @@ void ColorSelectorHexInputWidget::set_color(const QColor& color) {
   }
 }
 
-connection ColorSelectorHexInputWidget::connect_color_signal(
+connection ColorSelectorHexInputWidget::connect_modified_signal(
     const ColorSignal::slot_type& slot) const {
-  return m_color_signal.connect(slot);
+  return m_modified_signal.connect(slot);
+}
+
+connection ColorSelectorHexInputWidget::connect_selected_signal(
+    const ColorSignal::slot_type& slot) const {
+  return m_selected_signal.connect(slot);
 }
 
 bool ColorSelectorHexInputWidget::eventFilter(QObject* watched,
     QEvent* event) {
-  if(watched == m_text_input && event->type() == QEvent::KeyPress) {
+  if(event->type() == QEvent::KeyPress) {
     auto e = static_cast<QKeyEvent*>(event);
-    if(e->key() == Qt::Key_Enter || e->key() == Qt::Key_Return) {
+    if(e->key() == Qt::Key_Enter || e->key() == Qt::Key_Return ||
+        e->key() == Qt::Key_Space) {
       auto text = m_text_input->text();
       if(text.length() == 6 || text.length() == 3) {
         auto color = QColor();
         color.setNamedColor(QString("#%1").arg(m_text_input->text()));
         m_color_name = color_name(color);
-        m_color_signal(color);
+        m_selected_signal(color);
```

Comment:
Only signal once you've committed all state. A signal introduces re-entrancy so any state not committed before the signal can end up being inconsistent afterwards.